src/Microsoft.DncEng.SecretManager/StorageTypes/AzureKeyVault.cs

```diff
@@ -98,5 +105,33 @@ public override async Task SetSecretValueAsync(AzureKeyVaultParameters parameter
             properties.ExpiresOn = value.ExpiresOn;
             await client.UpdateSecretPropertiesAsync(properties);
         }
+
+        public override async Task EnsureKeyAsync(AzureKeyVaultParameters parameters, string name, SecretManifest.Key config)
+        {
+            var client = await CreateKeyClient(parameters);
+            KeyVaultKey key = null;
+            try
+            {
+                key = await client.GetKeyAsync(name);
```

Comment:
Should this just return after getting it? (Probably with a comment).  Took me a short bit to realize this assignment wasn't actually interesting, and only the side-effect of this method call (the RequestFailedException) was actually being used.

It could just be "await client.GetKeyAsync(name); return; // Since the key exists, we have 'Ensured' it, so we are done"

Reply:
Ahh yea, that works better. I'll do that.